src/explore-education-statistics-admin/src/components/NavigableSections.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ReactNode } from 'react';
+import PreviousNextLinks from '@admin/components/PreviousNextLinks';
+import Link from '@admin/components/Link';
+
+export interface Section<SectionType extends {}> {
+  section: SectionType;
+  label: string;
+  linkTo: string;
+}
+
+interface Props<SectionType> {
+  navigationHeadingText: string;
+  navigationHeadingSubtitle: string;
+  availableSections: Section<SectionType>[];
+  selectedSection: Section<SectionType>;
+  children: ReactNode;
+}
+
+/**
+ * Represents a portion of a page with navigable tabbed sections, and Next / Previous buttons to switch between the
+ * sections.  A client of this component provides a full ordered list of the available sections, and the currently
+ * selected section.  From this, the Previous and Next links are inferred.
+ *
+ * The client also provides the appropriate body for the selected section, which is rendered within this component
+ * as its props.children.
+ *
+ * The SectionType generic type represents a type for a unique id per section, whatever works best for the client.
+ * This could for instance be an enum of available sections, or a string or number id.
+ *
+ * @param availableSections
+ * @param selectedSection
+ * @param children
+ * @constructor
+ */
+const NavigableSections = <SectionType extends {}>({
+  availableSections,
+  selectedSection,
+  children,
+}: Props<SectionType>) => {
+  const nextSection =
+    availableSections.indexOf(selectedSection) < availableSections.length - 1
+      ? availableSections[availableSections.indexOf(selectedSection) + 1]
+      : null;
+
+  const previousSection =
+    availableSections.indexOf(selectedSection) > 0
+      ? availableSections[availableSections.indexOf(selectedSection) - 1]
+      : null;
+
+  return (
+    <>
+      <MainSectionNavigation
+        selectedSection={selectedSection.section}
+        availableSections={availableSections}
+      />
+      {children}
+      <PreviousNextLinks
+        previousSection={previousSection || undefined}
+        nextSection={nextSection || undefined}
+      />
+    </>
+  );
+};
+
+interface NavigationProps<SectionType> {
+  availableSections: Section<SectionType>[];
+  selectedSection: SectionType;
+}
+
+const MainSectionNavigation = <SectionType extends {}>({
+  availableSections,
+  selectedSection,
+}: NavigationProps<SectionType>) => {
+  return (
+    <>
+      <nav className="app-navigation govuk-!-margin-bottom-9">
```

Comment:
Two things wrong here:

- We should be using a Sass module instead of global styles.
- It looks really broken on mobile.

Reply:
Yup that's fine.  I was actually going to work with Marv to generate some sass modules for the work done so far, as an additional piece of work when he's done updating prototypes.  As this is a WIP at the moment and not heading back onto master just yet, I just wanted to get this initial PR in so as not to create a monster later on :)

Reply:
Cool cool as long as it's revisited, otherwise I would just do it here as converting to a Sass module isn't particularly hard. It's mostly just copy and pasting and changing class names to use camelCase.

Reply:
Will raise a ticket for this

Reply:
Created https://alpha-dfe-data.atlassian.net/browse/DFE-969